src/seedsigner/views/view.py

```diff
@@ -242,30 +242,41 @@ def run(self):
             return Destination(PowerOffView)
 
 
-
+@dataclass
 class RestartView(View):
+    is_screenshot_renderer: bool = False
+
     def run(self):
         from seedsigner.gui.screens.screen import ResetScreen
-        thread = RestartView.DoResetThread()
+        thread = RestartView.DoResetThread(is_screenshot_renderer=self.is_screenshot_renderer)
         thread.start()
         self.run_screen(ResetScreen)
 
 
     class DoResetThread(BaseThread):
+        def __init__(self, is_screenshot_renderer: bool = False):
+            self.is_screenshot_renderer = is_screenshot_renderer
+            super().__init__()
+
+
         def run(self):
+            import os
+            import sys
             import time
-            from subprocess import call
 
             # Give the screen just enough time to display the reset message before
             # exiting.
             time.sleep(0.25)
+            if self.is_screenshot_renderer:
+                # For the screenshot generator, we don't actually want to restart
+                return
+
```

Comment:
A debugging print out or log would be nice here to say that we're doing the restart.